solaris/Platform.c

```diff
@@ -267,16 +267,21 @@ static int Platform_buildenv(void* accum, struct ps_prochandle* Phandle, uintptr
    envAccum* accump = accum;
    (void) Phandle;
    (void) addr;
+
    size_t thissz = strlen(str);
-   if ((thissz + 2) > (accump->capacity - accump->size)) {
-      accump->env = xRealloc(accump->env, accump->capacity *= 2);
-   }
-   if ((thissz + 2) > (accump->capacity - accump->size)) {
-      return 1;
+
+   while ((thissz + 2) > (accump->capacity - accump->size)) {
+      if (accump->capacity > (SIZE_MAX >> 1))
```

Comment:
Please use Integer arithmetic not bit shifting. Why do you limit to SIZE_MAX >> 1?

Reply:
Limit due to overflow in multiplication below otherwise …